wordpress-mobile/WordPress-Android
# Improve quota snackbar

Fixes #7103 

This PR changes the snackbar message when the quota is exceeded and adds a light snackbar in the dark mode. I've removed the `WPSnackbar` because it should no longer be necessary. We can use the standard component which already uses material design. 

I've added a snackbar to the editor which is shown when the media upload fails. The message is now:
"Media upload failed.
Failure reasion". 
It looks like this: 

![Screenshot_1610983160](https://user-images.githubusercontent.com/1079756/104933549-58daa000-59a9-11eb-9264-5e3a1fbbd760.png)

Could you check the light snackbar designs to see if they look alright @SylvesterWilmott ? I've used the standard styles with the action text color being the secondary color (pink) with no alpha instead of the standard primary (in our case blue) with 0.5 alpha. 

![Screenshot_1610981211](https://user-images.githubusercontent.com/1079756/104933706-86bfe480-59a9-11eb-8e75-fef68209ff43.png)
<img width="619" alt="Screenshot 2021-01-18 at 16 25 38" src="https://user-images.githubusercontent.com/1079756/104933842-b0790b80-59a9-11eb-8f8d-e2b92842aee7.png">
<img width="623" alt="Screenshot 2021-01-18 at 16 27 35" src="https://user-images.githubusercontent.com/1079756/104934042-f504a700-59a9-11eb-90fc-943c0e711a28.png">
<img width="628" alt="Screenshot 2021-01-18 at 16 27 14" src="https://user-images.githubusercontent.com/1079756/104934048-f635d400-59a9-11eb-81d9-40f4a9432076.png">



To test (both in light and dark mode):
- Trigger a snackbar without an action (for example by updating a post)
- Check it looks as expected
- Trigger a snackbar with an action
- Check it looks as expected
- Click on the action
- Check it works as expected


To test: 
- Try to add an image from the editor on a site with full quota
- Notice the snackbar is visible
- Notice the message is updated

PR submission checklist:
